sql/sql_tablespace.cc

```diff
@@ -453,6 +453,13 @@ bool Sql_cmd_create_tablespace::execute(THD *thd) {
     return true;
   }
 
+  // Acquire Percona's LOCK TABLES FOR BACKUP lock
+  if (thd->backup_tables_lock.abort_if_acquired() ||
+      thd->backup_tables_lock.acquire_protection(
+          thd, MDL_TRANSACTION, thd->variables.lock_wait_timeout)) {
+    return true;
+  }
+
```

Comment:
Where is upstream (LIFB) taken? It is safe to take at the same place. Also, the order is important. Either LIFB first and then LTFB or LTFB first and then LIFB. Please check this order in other places (like undo truncation for example)

Reply:
It is taken at [line](https://github.com/percona/percona-server/blob/3d545ea8f8e72d3c59b5342788ca98170a6b09b2/sql/sql_tablespace.cc#L246) inside  lock_tablespace_names .
But undo it done [at](https://github.com/percona/percona-server/blob/3d545ea8f8e72d3c59b5342788ca98170a6b09b2/sql/sql_tablespace.cc#L1399)

for undo and my patch the order is sam, we  first we try to take [LTFB](https://github.com/percona/percona-server/blob/3d545ea8f8e72d3c59b5342788ca98170a6b09b2/sql/sql_tablespace.cc#L1382) and then [LIFB](https://github.com/percona/percona-server/blob/3d545ea8f8e72d3c59b5342788ca98170a6b09b2/sql/sql_tablespace.cc#L1399)

Reply:
should I try moving LTFB also inside lock_tablespace_names?

Reply:
global read lock is FTWRL right?

Reply:
yes.